### ISSUE TITLE
Pimcore X: Webdav doesn't work correctly (Exception)

### ISSUE THREAD

**Issue by akno77** [pimcore/pimcore#9175]
# Bug Report

### Expected behavior

Open WebDAV https://demo.pimcore.com/admin/asset/webdav
If you click a folder the content is displayed. If you click a file, the file is displayed.

### Actual behavior

If you open a folder like https://demo.pimcore.com/admin/asset/webdav/Brand%20Logos/ the following error message is displayed:
**4.1.5 Sabre\DAV\Exception\NotFound File not found: rand Logos**

The same behavior is with files.

### Steps to reproduce

1. Open WebDav via Browser: https://demo.pimcore.com/admin/asset/webdav
2. Click the "Brand Logos" folder => Error Message
3. You see in the error message, that from the folder Name "Brand Logos" the first letter is stripped. If you add any letter before "Brand Logo" like https://demo.pimcore.com/admin/asset/webdav/XBrand%20Logos/ it works!

### Code example
```php
// your code
```

# Feature Request

**Comment by craigatkinson** [comment]
I took a brief look and the change/reason is the commit:
#8727 

Specifically the change to getChild here:
https://github.com/pimcore/pimcore/blob/4550a053fac47a1d2c0406bf15c9799ef078bbb7/models/Asset/WebDAV/Folder.php#L82

`substr($name, strrpos($name, '/') + 1`

Will strip the first character, even if it's not a '/'


The former method:

```
$nameParts = explode('/', $name);
  echo      $nameParts[count($nameParts) - 1];
```

Will only strip '/', otherwise it returns correctly the full filename.


It seems stripping that / is not necessary. It's already happening somewhere else. I tried to look into SabreDAV but it doesn't seem to be done there.

**Comment by brusch** [comment]
Fixed by #9224
